### ISSUE TITLE
Koodikatselmointi

### ISSUE THREAD

**Issue by alexalgrund** [psangi-hy/ot-harjoitustyo#1]
Peli on ajettavissa ja asennettavissa annetuilla ohjeilla kuten kuuluukin. Luokkien rakenne on sinällään hyvin selkeä ja tämä on yhdenmukaista pelin yksinkertaisen idean kanssa. Vaikuttaa siis siltä, että toiminnallisuuden rakenne on juuri sopivissa mittasuhteissa, eikä asiota ole yritetty tehdä vaikeamman kautta tai toistamalla turhaan jo olemassaolevaa koodia. Myös muut invoke-komennot ovat ajettavissa. Vaatimusmääritelmän perusteella isompi osa pelin toiminnallisuuksista lienee vielä työn alla ja tältä kannalta olisi tärkeää liittää eri tiedostojen funktioihin ja luokkiin kuvauksia niitten toiminnallisuudesta. Myös tiedostot `graphical_ui.py` ja `text_ui.py` ovat rakenteeltaan vähän pitkiä ja niiden jakamista luokkiin kannattanee harkita projektin loppuvaiheessa. Muistathan myös, että ohjelmakoodiin kirjoitettavat kuvaukset on tarkoitus kirjoittaa tällä kurssilla englanniksi, vaikka dokumentaatio voikin muuten olla suomeksi. Mielestäni pelissä tulisi olla myös ilmoituksia pelaajalle kuten "GAME OVER" pelin päättyessä. Tällä hetkellä pelin päättyessä pelaajan täytyy sulkea ikkuna oikean yläkulman raksista. Ymmärrän toki tämän jossain määrin pelin yksikertaisuuden ja idean tunnettavuuden takia, mutta kehittäisin ainakin itse vielä lisää pelikokemuksen intuitiivisuutta. Yleisesti ottaen ohjelmakoodi on kuitenkin jaettu osiinsa hyvin.

Peli on ladattu arviotavaksi 10.12.2024 klo. 15.00

**Comment by psangi-hy** [comment]
No niin, luin tämän juuri, kun olin kirjoittanut kaikki docstringit suomeksi. :D Osaatko sanoa, missä tarkalleen lukee, että koodi pitää kommentoida englanniksi?
Kiitos hienosta palautteesta. :)

**Comment by alexalgrund** [comment]
Aloitusluennon kalvoissa oli mainittu tällä tavalla, että englanniksi :D Hauskaa, että joku lukee nää ihan sillä ajatuksella, että syntyy tämmöstä keskustelua. Ei oo ennnen ollu tämmöstä, että keskustelemaan aletaan. :D En tosin oon varma tarkoittaako toi nyt lähinnä tiedostomuodoissa olevia laajempia dokumentteja, mutta näin siis itse tein. Mä tulkitsin asiaa niin, että kaikki kooditiedostoissa oleva teksti englanniksi ja muuten suomeksi.

![Screenshot from 2024-12-10 21-31-07](https://github.com/user-attachments/assets/620d00af-2265-40c3-ac5b-68eb4e9ecfbc)

**Comment by psangi-hy** [comment]
No, koska tossa mainitaan nimenomaan muuttujat, luokat ja metodit, niin mä tulkitsen tän niin, että kommentit lasketaan kuitenkin dokumentaatioksi, joka saa olla myös suomeksi. Mutta toivottavasti viimeistään Labtoolissa tulee sitten palautetta, jos asia on toisin. Kiitti, kun viitsit etsiä mulle ton kalvon. :)
